MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs

```diff
@@ -1018,117 +964,119 @@ private bool IsClose(double d1, double d2, double variance = .005)
+            if (!string.IsNullOrEmpty(bps)
+                && int.TryParse(bps, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parsedBps))
+            {
+                return parsedBps;
             }
 
             return null;
         }
 
         private double? GetRuntimeSecondsFromTags(MediaStreamInfo streamInfo)
         {
-            if (streamInfo != null && streamInfo.Tags != null)
+            if (streamInfo?.Tags == null)
             {
-                var duration = GetDictionaryValue(streamInfo.Tags, "DURATION-eng") ?? GetDictionaryValue(streamInfo.Tags, "DURATION");
-                if (!string.IsNullOrEmpty(duration) && TimeSpan.TryParse(duration, out var parsedDuration))
-                {
-                    return parsedDuration.TotalSeconds;
-                }
+                return null;
+            }
+
+            var duration = GetDictionaryValue(streamInfo.Tags, "DURATION-eng") ?? GetDictionaryValue(streamInfo.Tags, "DURATION");
+            if (!string.IsNullOrEmpty(duration) && TimeSpan.TryParse(duration, out var parsedDuration))
+            {
+                return parsedDuration.TotalSeconds;
             }
 
             return null;
         }
 
         private long? GetNumberOfBytesFromTags(MediaStreamInfo streamInfo)
         {
-            if (streamInfo != null && streamInfo.Tags != null)
+            if (streamInfo?.Tags == null)
             {
-                var numberOfBytes = GetDictionaryValue(streamInfo.Tags, "NUMBER_OF_BYTES-eng") ?? GetDictionaryValue(streamInfo.Tags, "NUMBER_OF_BYTES");
-                if (!string.IsNullOrEmpty(numberOfBytes)
-                    && long.TryParse(numberOfBytes, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parsedBytes))
-                {
-                    return parsedBytes;
-                }
+                return null;
+            }
+
+            var numberOfBytes = GetDictionaryValue(streamInfo.Tags, "NUMBER_OF_BYTES-eng")
+                                ?? GetDictionaryValue(streamInfo.Tags, "NUMBER_OF_BYTES");
+            if (!string.IsNullOrEmpty(numberOfBytes)
+                && long.TryParse(numberOfBytes, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parsedBytes))
+            {
+                return parsedBytes;
             }
 
             return null;
         }
 
         private void SetSize(InternalMediaInfoResult data, MediaInfo info)
         {
-            if (data.Format != null)
+            if (data.Format == null)
             {
-                if (!string.IsNullOrEmpty(data.Format.Size))
-                {
-                    info.Size = long.Parse(data.Format.Size, _usCulture);
-                }
-                else
-                {
-                    info.Size = null;
-                }
+                return;
             }
+
+            info.Size = !string.IsNullOrEmpty(data.Format.Size)
```

Comment:
I generally prefer not having negations. It flows better in my mind if I don't have to negate an expression.

Reply:
Agreed, unfortunately there's no inverse method for this I fear.

Reply:
You could just not negate it :P

```C#
             info.Size = string.IsNullOrEmpty(data.Format.Size)
                ? null
                : long.Parse(data.Format.Size, _usCulture);
```

Reply:
Right, but I personally think the common/normal case should be handled first, will fallback/error cases second.